### ISSUE TITLE
Proposal: Make the xclipboard layer to use system clipboard for copy/paste

### ISSUE THREAD

**Issue by xuhdev** [syl20bnr/spacemacs#16745]
Currently, the `xclipboard` layer provides `spacemacs/xclipboard-copy` and
`spacemacs/xclipboard-paste` for copying and pasting the active region. But I think
things can become more useful if we make the layer to copy/paste from the
clipboard by default with a function to switch it on/off, for all kinds of
copying/pasting.

I have a small script that works with wayland: https://gist.github.com/yorickvP/6132f237fbc289a45c808d8d75e0e1fb?permalink_comment_id=5345689#gistcomment-5345689

In a nutshell, it replaces `interprogram-cut-function` and
`interprogram-paste-function` to make use of the external clipboard program. It
should be easily adaptable to other commands.

What do you think?

**Comment by xuhdev** [comment]
There are too many quirks in clipboard behavior maybe it's not the time yet.
